crawler/file_processing.py

```diff
@@ -370,6 +393,37 @@ def set_state_for_file(self) -> CentreFileState:
 
         return self.file_state
 
+    def get_important_unprocessed_priority_samples(self, root_sample_ids: List[str]) -> List[Any]:
+        """
+        Description
+        check if sample is in priority_samples either must_sequence/preferentially_sequence is true, and processed false
+        Arguments:
+            x {Type} -- description
+        """
+        matching_priority_entry = {FIELD_ROOT_SAMPLE_ID: {"$in": root_sample_ids}}
+        unprocessed = {FIELD_PROCESSED: False}
+        of_importance = {"$or": [{FIELD_MUST_SEQUENCE: True}, {FIELD_PREFERENTIALLY_SEQUENCE: True}]}
+
+        query = {"$and": [matching_priority_entry, unprocessed, of_importance]}
+
+        priority_samples_collection = get_mongo_collection(self.get_db(), COLLECTION_PRIORITY_SAMPLES)
+
+        priority_sample_cursor = priority_samples_collection.find(query)
+        return list(priority_sample_cursor)
+
+    def get_root_sample_ids_from_samples_collection(self, mongo_ids: List[str]) -> List[str]:
+        """
+        Get the root sample ids for samples with the given mongo _id
+        Arguments:
+            x {Type} -- description
+        """
+
+        def extract_root_sample_id(sample: SampleDoc) -> SampleDocValue:
+            return sample[FIELD_ROOT_SAMPLE_ID]
+
+        samples_collection = get_mongo_collection(self.get_db(), COLLECTION_SAMPLES)
+        return list(map(extract_root_sample_id, samples_collection.find({FIELD_MONGODB_ID: {"$in": mongo_ids}})))
```

Comment:
This function may be obsolete

Reply:
both get_root_sample_ids_from_samples_collection and extract_root_sample_id should be